File: dart/src/main_test.dart
import 'main.dart';

void main() {
  final calculator = Calculator();

  final addResult = calculator.add(2, 3);
  assert(addResult == 5, 'Expected 5');

  final divideResult = calculator.divide(10, 2);
  assert(divideResult == 5.0, 'Expected 5.0');

  bool caught = false;
  try {
    calculator.divide(1, 0);
  } on ArgumentError {
    caught = true;
  }
  assert(caught, 'Should have thrown');

  print('All tests passed!');
}

File: javascript/src/main.test.js
const assert = require('assert');

function testAdd() {
    assert.strictEqual(add(2, 3), 5);
}

function testDivide() {
    assert.strictEqual(divide(10, 2), 5);
}

function testDivideByZero() {
    try {
        divide(1, 0);
        assert.fail('Should have thrown');
    } catch (err) {
        assert.strictEqual(err.message, 'Division by zero');
    }
}

testAdd();
testDivide();
testDivideByZero();
console.log('All tests passed!');